frontend/packages/dev-console/src/components/import/serverlessfunc/FuncSection.tsx

```diff
@@ -0,0 +1,96 @@
+import { BuilderImage } from '@console/dev-console/src/utils/imagestream-utils';
+import { getGitService } from '@console/git-service/src';
+import { evaluateFunc } from '@console/git-service/src/utils/serverless-strategy-detector';
+import { Loading } from '@console/internal/components/utils';
+import BuilderImageTagSelector from '../builder/BuilderImageTagSelector';
+import { Resources, Runtime } from '../import-types';
+import { useResourceType } from '../section/useResourceType';
+import { getRuntimeImage } from './func-utils';
+import './FuncSection.scss';
+
+const FuncSection = ({ builderImages }) => {
+  const { t } = useTranslation();
+  const { values, setFieldValue, setFieldError, errors } = useFormikContext<FormikValues>();
+  const {
+    git: { url, type, ref, dir, secretResource },
+    image,
+  } = values;
+  const [runtimeImage, setRuntimeImage] = React.useState<BuilderImage>();
+  const [loaded, setLoaded] = React.useState<boolean>(false);
+  const [, setResourceType] = useResourceType();
+
+  React.useEffect(() => {
+    const gitService = url && getGitService(url, type, ref, dir, secretResource);
+    gitService &&
+      evaluateFunc(gitService)
+        .then((res) => {
+          setResourceType(Resources.KnativeService);
+          setRuntimeImage(getRuntimeImage(res.values.runtime as Runtime, builderImages));
+          setFieldValue('resources', Resources.KnativeService);
+          setFieldValue('build.env', res.values.builderEnvs);
+          setFieldValue('deployment.env', res.values.runtimeEnvs);
+        })
+        .catch((err) => {
+          // eslint-disable-next-line no-console
+          console.warn('Error fetching Serverless Function YAML: ', err);
+          setFieldError('ServerlessFunction', err.message);
+        })
+        .finally(() => setLoaded(true));
+  }, [
+    setFieldValue,
+    setFieldError,
+    url,
+    type,
+    ref,
+    dir,
+    secretResource,
+    setResourceType,
+    builderImages,
+  ]);
+
+  React.useEffect(() => {
+    if (loaded && runtimeImage) {
+      setFieldValue('image.tag', runtimeImage?.recentTag?.name);
+      setFieldValue('image.selected', runtimeImage?.name);
+      setFieldValue('image.recommended', runtimeImage?.name);
+    }
+  }, [runtimeImage, setFieldValue, loaded]);
+
+  React.useEffect(() => {
+    if (loaded && !runtimeImage) {
+      setFieldError('ServerlessFunction', 'Unsupported Runtime detected');
+    }
+  }, [setFieldError, loaded, runtimeImage, errors]);
+
+  if (loaded && !runtimeImage) {
+    return (
+      <FormSection>
+        <Alert
+          className="odc-func-strategy-section__error-alert"
+          isInline
+          variant="danger"
+          title={t('devconsole~Import is not possible.')}
+        >
+          {t(
+            'devconsole~Unsupported Runtime detected. Please update the Repository URL or change the Build Strategy to continue.',
```

Comment:
Not getting this alert for unsupported runtime like rust https://github.com/vikram-raj/hello-func-rust